src/azul/downloader.py

```diff
@@ -10,139 +10,40 @@
-        files.
-
-        :param bundle_uuid: The bundle to pull from the DSS
-        :param bundle_version: The bundle version to pull from the DSS
-        :param replica: The replica which we should be pulling from
-        (e.g aws, gcp, etc)
-        :return: Returns a tuple separating (metadata_files, data_files)
-        """
-        # The blue box may have trouble returning bundle, so __attempt 3 times
-        log.info("Getting bundle %s.%s from DSS.", bundle_uuid, bundle_version)
-        bundle = self.__attempt(3,
-                                self.dss_client.get_bundle,
-                                SwaggerAPIException,
-                                uuid=bundle_uuid,
-                                version=bundle_version,
-                                replica=replica)
-        # Separate files in bundle by metadata files and data files
-        _files = bundle['bundle']['files']
-        metadata_files = {f["name"]: f for f in _files if f["indexed"]}
-        manifest = _files
-        # Return as a tuple
-        return metadata_files, manifest
-
-    def __get_file(self, file_uuid, file_version, replica):
-        """
-        Get a file from the Blue Box.
-
-        This function gets a file from the blue box based on
-        the 'file_uuid'. It will attempt to get the file three
-        times.
-
-        :param file_uuid: Specifies which file to get
-        :param file_version: Specifies which file to get
-        :param replica: Specifies the replica to pull from
-        :return: Contents of that file
-        """
-        # The blue box may have trouble returning file, so __attempt 3 times
-        log.info("Getting file %s version %s from DSS.", file_uuid, file_version)
-        _file = self.__attempt(3,
-                               self.dss_client.get_file,
-                               SwaggerAPIException,
-                               uuid=file_uuid,
-                               version=file_version,
-                               replica=replica)
-        return _file
-
-    def extract_bundle(self, request, replica="aws"):
+    def download_bundle_metadata(self, uuid, version):
         """
         Get the files and actual metadata.
 
         This is the main method that will extract the contents of the bundle
         and separate it into a tuple of (metadata_files, data_files), where
         the metadata_files are actual contents of the metadata files and the
         data_files are the metadata describing the files.
-
-        :param request: The contents of the DSS event notification
-        :param replica: The replica to which pull the bundle from
         """
-        bundle_uuid = request['match']['bundle_uuid']
-        bundle_version = request['match']['bundle_version']
-        metadata_files, manifest = self.__get_bundle(bundle_uuid, bundle_version, replica)
-        num_workers = int(os.environ['AZUL_DSS_WORKERS'])
```

Comment:
This configuration should be retained. `download_bundle_metadata` has a `num_workers` argument that you can use. You may have to upgrade the dependency to get that argument.